ADBench/run-all.ps1

```diff
@@ -169,26 +173,48 @@ function run_command ($indent, $outfile, $timeout, $cmd) {
     $ProcessInfo.Arguments = $args
     $Process = New-Object System.Diagnostics.Process
     $Process.StartInfo = $ProcessInfo
+
     try {
         $Process.Start() | Out-Null
     } catch {
         write-error "Failed to start process $cmd $args"
         throw "failed"
     }
-    $status = $Process.WaitForExit($timeout * 1000)
-    if ($status) {
-        $stdout = $Process.StandardOutput.ReadToEnd().Trim().Replace("`n", "`n${indent}stdout> ")
-        $stderr = $Process.StandardError.ReadToEnd().Trim().Replace("`n", "`n${indent}stderr> ")
-        $allOutput = "${indent}stdout> " + $stdout + "`n${indent}stderr> " + $stderr
-        Write-Host "$allOutput"
-        return [RunCommandStatus]::Finished
-    } else {
-        $Process.Kill()
-        Write-Host "${indent}Killed after $timeout seconds"
-        Store-NonFatalError "Process killed after $timeout seconds`n[$cmd $args]"
-        create_timeout_file $outfile $timeout
-        return [RunCommandStatus]::Timeout
+
+    $OOMCheckingInterval = 500      # milliseconds
+    while (-not $Process.WaitForExit($OOMCheckingInterval)) {
+        $Process.Refresh()
+        if (-not $Process.HasExited) {
+            $mem = 0
+            try {
+                $mem = $Process.PeakPagedMemorySize64
+            } catch {
+                # nothing to do because the process can exit by itself
+            }
+
+            $runTime = ((Get-Date) - $Process.StartTime).Duration().TotalSeconds
+
+            if ($runTime -ge $timeout) {
+                $Process.Kill()
+                Write-Host "${indent}Killed after $timeout seconds"
+                Store-NonFatalError "Process killed after $timeout seconds`n[$cmd $args]"
+                create_timeout_file $outfile $timeout
+                return [RunCommandStatus]::Timeout
+            } elseif ($mem -ge $max_memory_amount_in_bytes) {
+                $Process.Kill()
+                Write-Host "${indent}Killed due to consuming $mem bytes of operating memory"
+                Store-NonFatalError "Process killed due to consuming $mem bytes of operating memory`n[$cmd $args]"
+                return [RunCommandStatus]::OutOfMemory
```

Comment:
I think, displaying `$mem` in GB here would be more convenient for users.

Reply:
It makes sense

Reply:
Done